pkg/clients/dns/dns.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	return res
+}
+
+func newSubResourceParameters(e *[]v1alpha1.SubResource) *[]dns.SubResource {
+	endpoints := make([]dns.SubResource, len(*e))
+
+	for i, s := range *e {
+		endpoints[i] = dns.SubResource{
+			ID: s.ID,
+		}
+	}
+
+	return &endpoints
+}
+
+// ZoneIsUpToDate decides if an upgrade is needed.
+func ZoneIsUpToDate(r *v1alpha1.Zone, az dns.Zone) bool {
+	up := NewZoneParameters(r)
+	if !cmp.Equal(up.Tags, az.Tags) || !cmp.Equal(up.RegistrationVirtualNetworks, az.RegistrationVirtualNetworks) ||
+		!cmp.Equal(up.ResolutionVirtualNetworks, az.ResolutionVirtualNetworks) {
+		return false
+	}
+
+	return true
+}
+
+// RecordSetAPI represents the API interface for a DNS RecordSet client
+type RecordSetAPI interface {
+	Get(ctx context.Context, r *v1alpha1.RecordSet) (dns.RecordSet, error)
+	CreateOrUpdate(ctx context.Context, r *v1alpha1.RecordSet) error
+	Delete(ctx context.Context, r *v1alpha1.RecordSet) error
+}
+
+// RecordSetClient is the concrete implementation of the ZoneAP interface for RecordSet that calls Azure API.
+type RecordSetClient struct {
+	dns.RecordSetsClient
+}
+
+// NewRecordSetClient creates and initializes a RecordSet instance.
+func NewRecordSetClient(cl dns.RecordSetsClient) *RecordSetClient {
+	return &RecordSetClient{
+		RecordSetsClient: cl,
+	}
+}
+
+// Get retrieves the requested DNS RecordSet
+func (c *RecordSetClient) Get(ctx context.Context, r *v1alpha1.RecordSet) (dns.RecordSet, error) {
+	return c.RecordSetsClient.Get(ctx, r.Spec.ForProvider.ResourceGroupName, r.Spec.ForProvider.ZoneName, meta.GetExternalName(r),
+		dns.RecordType(r.Spec.ForProvider.RecordType))
+}
+
+// CreateOrUpdate creates or updates a DNS RecordSet
+func (c *RecordSetClient) CreateOrUpdate(ctx context.Context, r *v1alpha1.RecordSet) error {
+	_, err := c.RecordSetsClient.CreateOrUpdate(ctx, r.Spec.ForProvider.ResourceGroupName, r.Spec.ForProvider.ZoneName,
+		meta.GetExternalName(r), dns.RecordType(r.Spec.ForProvider.RecordType), NewRecordSetParameters(&r.Spec.ForProvider),
+		"", "")
+
+	return err
+}
+
+// Delete deletes the given DNS RecordSet
+func (c *RecordSetClient) Delete(ctx context.Context, r *v1alpha1.RecordSet) error {
+	_, err := c.RecordSetsClient.Delete(ctx, r.Spec.ForProvider.ResourceGroupName, r.Spec.ForProvider.ZoneName,
+		meta.GetExternalName(r), dns.RecordType(r.Spec.ForProvider.RecordType), r.Status.AtProvider.Etag)
+
+	return err
+}
+
+// UpdateRecordSetStatusFromAzure updates the status related to the external
+// Azure DNS RecordSet in the RecordSetStatus
+func UpdateRecordSetStatusFromAzure(v *v1alpha1.RecordSet, az dns.RecordSet) {
+	v.Status.AtProvider.ID = azure.ToString(az.ID)
+	v.Status.AtProvider.Etag = azure.ToString(az.Etag)
+	v.Status.AtProvider.Name = azure.ToString(az.Name)
+	v.Status.AtProvider.Type = azure.ToString(az.Type)
+	v.Status.AtProvider.ProvisioningState = azure.ToString(az.ProvisioningState)
+	v.Status.AtProvider.FQDN = azure.ToString(az.Fqdn)
```

Comment:
Interesting that Azure SDK uses `Fqdn`. But per Crossplane conventions, I think `FQDN` is more appropriate. What do you think?

Reply:
I agree with you. I think FQDN is more appropriate.